app/directives/proposalList/proposalList.directive.js

```diff
@@ -258,10 +267,17 @@ angular.module('dashboard')
 
             $scope.$watch(function () {
                 return {
-                    guid: $scope.guid
+                    topic: $scope.topic
                 };
             }, function (data) {
-                getProposals(data.guid);
+                if (angular.isObject(data) && angular.isObject(data.topic)) {
```

Comment:
Was it so that new/old might be the same and in watch we should `angular.equals` just in case?